core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -143,6 +163,28 @@ void start() {
       return stateCache;
    }
 
+   @Override
+   public CompletableFuture<Void> createTemplate(String name, Configuration configuration, EnumSet<CacheContainerAdmin.AdminFlag> flags) {
+      Cache<ScopedState, Object> cache = getStateCache();
+      ScopedState key = new ScopedState(TEMPLATE_SCOPE, name);
+      return cache.containsKeyAsync(key).thenCompose(exists -> {
```

Comment:
`DefaultCacheManager` and `Configuration` don't allow a template and a cache to have the same name, shouldn't we do that here as well? Either way, the documentation should be clear about the namespaces and there should be a test.

It might have been simpler to keep a single scope and rely on `Configuration.isTemplate()` to distinguish between caches and templates.

Reply:
> It might have been simpler to keep a single scope and rely on `Configuration.isTemplate()` to distinguish between caches and templates.

This isn't possible as we don't serialize whether a configuration is a template or not. This is always set via the `ConfigurationBuilder`